src/api/request.c

```diff
@@ -159,12 +161,8 @@ void process_request(const char *client_message, int *sock)
 	{
 		processed = true;
 		logg("Received API request to re-resolve host names");
-		// Important: Don't obtain a lock for this request
-		//            Locking will be done internally when needed
-		// onlynew=false -> reresolve all host names
-		resolveClients(false);
-		resolveForwardDestinations(false);
-		logg("Done re-resolving host names");
+		// This will be done in the resolving thread
+		want_reresolve = true;
```

Comment:
Moving this into the resolver thread improved multi-core performance by lowering the time the data structure needs to be blocked.